plugins/input/bitbucket/server/server.go

```diff
@@ -1,57 +1,218 @@
 package server
 
 import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
 	"os"
 	"strings"
-	"context"
+	"time"
 
-	bitbucketv1 "github.com/gfleury/go-bitbucket-v1"
+	gjson "github.com/tidwall/gjson"
+	logrus "github.com/sirupsen/logrus"
+	// bitbucketv1 "github.com/gfleury/go-bitbucket-v1"
+
+	common "github.com/parinithshekar/gitsink/common"
+	config "github.com/parinithshekar/gitsink/common/config"
+	logger "github.com/parinithshekar/gitsink/wrap/logrus/v1"
 )
 
 // Server struct defines the data fields in bitbucket-server object
 type Server struct {
-	BaseURL    string
-	APIBaseURL string
-	accountID  string
-	api        *bitbucketv1.APIClient
+	apiBaseURL string
+
+	accountID   string
+	accessToken string
+	kind        string
+	api         *http.Client
+}
+
+// Credentials fetches amd returns the accountID and accessToken from environment variables
+func (server *Server) Credentials() (string, string) {
+	accountID := os.Getenv(server.accountID)
```

Comment:
Should: Error check if these env vars actually exist, if they are required.  This might be done with Kingpin.

Reply:
I'm checking if these env vars exist when the user tries to create a server object in the New() method lines 54-72. If the env var doesn't exist, the bitbucket-server object creation returns an error.